src/Core/src/Platform/Android/WrapperView.cs

```diff
@@ -83,12 +84,41 @@ protected override void OnMeasure(int widthMeasureSpec, int heightMeasureSpec)
 
 		public override void RequestLayout()
 		{
-			// Redraw shadow (if exists)
-			_invalidateShadow = true;
+			// Redraw shadow (if exists and required)
+			_invalidateShadow = ShouldInvalidateShadow();
 
 			base.RequestLayout();
 		}
 
+		bool ShouldInvalidateShadow()
+		{
+			if (_shadowBitmap == null)
+				return true;
+
+			if (_viewBounds == null)
+				return false;
+	
+			var viewHeight = _viewBounds.Height();
+			var viewWidth = _viewBounds.Width();
```

Comment:
Could we also consider making `_viewBounds` a regular MAUI `Rectangle`, whatever type that is? Just so we don't call into Java at all.